services/proxy/src/Proxy/API.hs

```diff
@@ -188,7 +188,7 @@ soundcloudStream url = do
     req1 <- Req.queryItem "client_id" s <$> Client.parseUrlThrow (Text.unpack url)
```

Comment:
How about not using `parseUrlThrow` at all?

```haskell
req <- maybe (failWith "invalid stream url") 
             (Req.noRedirect . Req.queryItem "client_id" s)
             (Client.parseRequest (Text.unpack url))
```

Otherwise we let it set a `checkResponse` handler only to remove it again some lines below.